pkg/common/utils_test.go

```diff
@@ -380,3 +383,127 @@ func TestGetConfigurationInt(t *testing.T) {
 		})
 	}
 }
+
+func TestZeroTimeInUnixNano(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    time.Time
+		expected *int64
+	}{
+		{
+			name:     "zero time",
+			input:    time.Time{},
+			expected: nil,
+		},
+		{
+			name:  "non-zero time",
+			input: time.Date(2024, time.June, 6, 0, 0, 0, 0, time.UTC),
+			expected: func() *int64 {
+				tInt := time.Date(2024, time.June, 6, 0, 0, 0, 0, time.UTC).UnixNano()
+				return &tInt
+			}(),
+		},
+		{
+			name:  "time in different timezone",
+			input: time.Date(2024, time.June, 6, 0, 0, 0, 0, time.FixedZone("UTC+8", 8*60*60)),
+			expected: func() *int64 {
+				tInt := time.Date(2024, time.June, 6, 0, 0, 0, 0, time.FixedZone("UTC+8", 8*60*60)).UnixNano()
+				return &tInt
+			}(),
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			result := ZeroTimeInUnixNano(tc.input)
+			if tc.expected == nil {
+				assert.Equal(t, tc.expected, result)
+			} else {
+				assert.Equal(t, *tc.expected, *result)
+			}
+		})
+	}
```

Comment:
Convert this to a series of checks. We have 3 cases and the readability is much better with this approach:

```
assert.Equal(t, ZeroTimeInUnixNano(time.Date(2024, time.June, 6, 0, 0, 0, 0, time.UTC)), time.Date(2024, time.June, 6, 0, 0, 0, 0, time.UTC).UnixNano())
...
```

Reply:
Sure. It looks better now!